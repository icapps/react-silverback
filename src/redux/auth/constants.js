export default {
  LOGIN_USER: 'LOGIN_USER',
  LOGIN_USER_PENDING: 'LOGIN_USER_PENDING',
  LOGIN_USER_REJECTED: 'LOGIN_USER_REJECTED',
  LOGIN_USER_FULFILLED: 'LOGIN_USER_FULFILLED',
  LOGOUT_USER: 'LOGOUT_USER',
};
