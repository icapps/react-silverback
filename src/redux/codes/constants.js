export default {
  GET_LANGUAGE_CODES: 'GET_LANGUAGE_CODES',
  GET_LANGUAGE_CODES_PENDING: 'GET_LANGUAGE_CODES_PENDING',
  GET_LANGUAGE_CODES_FULFILLED: 'GET_LANGUAGE_CODES_FULFILLED',
  GET_LANGUAGE_CODES_REJECTED: 'GET_LANGUAGE_CODES_REJECTED',

  GET_LANGUAGE_CODE_BY_ID: 'GET_LANGUAGE_CODE_BY_ID',
  GET_LANGUAGE_CODE_BY_ID_PENDING: 'GET_LANGUAGE_CODE_BY_ID_PENDING',
  GET_LANGUAGE_CODE_BY_ID_FULFILLED: 'GET_LANGUAGE_CODE_BY_ID_FULFILLED',
  GET_LANGUAGE_CODE_BY_ID_REJECTED: 'GET_LANGUAGE_CODE_BY_ID_REJECTED',

  CREATE_LANGUAGE_CODE: 'CREATE_LANGUAGE_CODE',
  CREATE_LANGUAGE_CODE_PENDING: 'CREATE_LANGUAGE_CODE_PENDING',
  CREATE_LANGUAGE_CODE_FULFILLED: 'CREATE_LANGUAGE_CODE_FULFILLED',
  CREATE_LANGUAGE_CODE_REJECTED: 'CREATE_LANGUAGE_CODE_REJECTED',

  DEPRECATE_LANGUAGE_CODE: 'DEPRECATE_LANGUAGE_CODE',
  DEPRECATE_LANGUAGE_CODE_PENDING: 'DEPRECATE_LANGUAGE_CODE_PENDING',
  DEPRECATE_LANGUAGE_CODE_FULFILLED: 'DEPRECATE_LANGUAGE_CODE_FULFILLED',
  DEPRECATE_LANGUAGE_CODE_REJECTED: 'DEPRECATE_LANGUAGE_CODE_REJECTED',

  UNDEPRECATE_LANGUAGE_CODE: 'UNDEPRECATE_LANGUAGE_CODE',
  UNDEPRECATE_LANGUAGE_CODE_PENDING: 'UNDEPRECATE_LANGUAGE_CODE_PENDING',
  UNDEPRECATE_LANGUAGE_CODE_FULFILLED: 'UNDEPRECATE_LANGUAGE_CODE_FULFILLED',
  UNDEPRECATE_LANGUAGE_CODE_REJECTED: 'UNDEPRECATE_LANGUAGE_CODE_REJECTED',

  GET_STATUS_CODES: 'GET_STATUS_CODES',
  GET_STATUS_CODES_PENDING: 'GET_STATUS_CODES_PENDING',
  GET_STATUS_CODES_REJECTED: 'GET_STATUS_CODES_REJECTED',
  GET_STATUS_CODES_FULFILLED: 'GET_STATUS_CODES_FULFILLED',

  SET_CODES_SORT: 'SET_CODES_SORT',
};
