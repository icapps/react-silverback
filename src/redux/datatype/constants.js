export default {
  GET_DATA_TYPE: 'GET_DATA_TYPE',
  GET_DATA_TYPE_PENDING: 'GET_DATA_TYPE_PENDING',
  GET_DATA_TYPE_REJECTED: 'GET_DATA_TYPE_REJECTED',
  GET_DATA_TYPE_FULFILLED: 'GET_DATA_TYPE_FULFILLED',
};
