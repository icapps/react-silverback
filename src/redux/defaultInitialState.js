const defaultInitialState = {
  isPending: false,
  isError: false,
  errorMessage: '',
};

export default defaultInitialState;
