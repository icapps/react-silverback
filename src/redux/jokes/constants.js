export default {
  GET_JOKE           : 'GET_JOKE',
  GET_JOKE_REQUEST   : 'GET_JOKE_REQUEST',
  GET_JOKE_FULFILLED : 'GET_JOKE_FULFILLED',
  GET_JOKE_REJECTED  : 'GET_JOKE_REJECTED',
};
