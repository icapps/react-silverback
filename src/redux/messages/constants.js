export default {
  SET_MESSAGE: 'SET_MESSAGE',
  
  CLEAR_MESSAGES: 'CLEAR_MESSAGES',
};
