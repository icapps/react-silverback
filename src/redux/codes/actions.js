import constants from './constants';
import { codesService } from '../../services';

export const getLanguageCodes = (page, limit, sortField, sortOrder, search) => {
  return {
    type: constants.GET_LANGUAGE_CODES,
    payload: codesService.getLanguageCodes(page, limit, sortField, sortOrder, search),
  };
};

export const getLanguageCodeById = id => {
  return {
    type: constants.GET_LANGUAGE_CODE_BY_ID,
    payload: codesService.getLanguageCodeById(id),
  };
};

export const createLanguageCode = languageCode => {
  return {
    type: constants.CREATE_LANGUAGE_CODE,
    payload: codesService.createLanguageCode(languageCode),
  };
};

export const deprecateLanguageCode = languageCodeId => {
  return {
    type: constants.DEPRECATE_LANGUAGE_CODE,
    payload: codesService.deprecateLanguageCode(languageCodeId),
  };
};

export const undeprecateLanguageCode = languageCodeId => {
  return {
    type: constants.UNDEPRECATE_LANGUAGE_CODE,
    payload: codesService.undeprecateLanguageCode(languageCodeId),
  };
};

export const setSort = (sortField, sortOrder) => {
  return {
    type: constants.SET_CODES_SORT,
    payload: {sortField, sortOrder},
  };
};

export const getStatusCodes = () => {
  return {
    type: constants.GET_STATUS_CODES,
    payload: codesService.getStatusCodes(),
  };
};