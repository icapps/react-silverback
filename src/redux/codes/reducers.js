import constants from './constants';
import defaultInitialState from '../defaultInitialState';
import { strings } from '../../utils/index';

const initialState = {
  languageCode: null,
  languageCodeList: [],
  languageCodeCount: 0,
  isCreatePending: false,
  isCreateError: false,
  ...defaultInitialState,
};

const errorMessageHandling = payload => payload.errors[0].title || strings.GENERAL_ERROR;

const codes = (state = initialState, action = {}) => {
  const { payload } = action;

  switch (action.type) {
    case constants.GET_LANGUAGE_CODES_FULFILLED:
      return {
        ...state,
        languageCodeList: payload.data,
        languageCodeCount: payload.meta.totalCount,
        languageCode: null,
        isPending: false,
      };
    case constants.GET_LANGUAGE_CODES_PENDING:
      return {
        ...state,
        isPending: true,
        isError: false,
        errorMessage: '',
      };
    case constants.GET_LANGUAGE_CODES_REJECTED:
      return {
        ...state,
        isPending: false,
        isError: true,
        errorMessage: errorMessageHandling(payload),
      };

    case constants.CREATE_LANGUAGE_CODE_FULFILLED:
      return {
        ...state,
        isCreatePending: false,
        languageCode: payload.data,
      };
    case constants.CREATE_LANGUAGE_CODE_PENDING:
      return {
        ...state,
        isCreatePending: true,
        isCreateError: false,
        errorMessage: '',
      };
    case constants.CREATE_LANGUAGE_CODE_REJECTED:
      return {
        ...state,
        isCreatePending: false,
        isCreateError: true,
        errorMessage: errorMessageHandling(payload),
      };

    case constants.DEPRECATE_LANGUAGE_CODE_FULFILLED:
      return {
        ...state,
        isPending: false,
      };
    case constants.DEPRECATE_LANGUAGE_CODE_PENDING:
      return {
        ...state,
        isPending: true,
        isError: false,
        errorMessage: '',
      };
    case constants.DEPRECATE_LANGUAGE_CODE_REJECTED:
      return {
        ...state,
        isPending: false,
        isError: true,
        errorMessage: errorMessageHandling(payload),
      };

    case constants.UNDEPRECATE_LANGUAGE_CODE_FULFILLED:
      return {
        ...state,
        isPending: false,
      };
    case constants.UNDEPRECATE_LANGUAGE_CODE_PENDING:
      return {
        ...state,
        isPending: true,
        isError: false,
        errorMessage: '',
      };
    case constants.UNDEPRECATE_LANGUAGE_CODE_REJECTED:
      return {
        ...state,
        isPending: false,
        isError: true,
        errorMessage: errorMessageHandling(payload),
      };

    case constants.GET_LANGUAGE_CODE_BY_ID_FULFILLED:
      return {
        ...state,
        languageCode: payload.data,
        isPending: false,
      };
    case constants.GET_LANGUAGE_CODE_BY_ID_PENDING:
      return {
        ...state,
        isPending: true,
        isError: false,
        errorMessage: '',
      };
    case constants.GET_LANGUAGE_CODE_BY_ID_REJECTED:
      return {
        ...state,
        isPending: false,
        isError: true,
        errorMessage: errorMessageHandling(payload),
      };
    
    //SET_SORT
    case constants.SET_CODES_SORT:
      const { sortField, sortOrder } = payload;
      return {
        ...state,
        sortField,
        sortOrder,
      };

    default:
      return state;
  }
};

export default codes;
