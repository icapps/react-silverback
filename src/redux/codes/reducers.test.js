import codes from './reducers';
import constants from './constants';
import defaultInitialState from '../defaultInitialState';

const initialState = {
  languageCodeList: [],
  languageCodeCount: 0,
  ...defaultInitialState,
};

describe('codes reducer', () => {
  it('should return the initial state', () => {
    expect(codes(undefined, {})).toMatchSnapshot();
  });
  it('should GET_LANGUAGE_CODES_FULFILLED', () => {
    expect(
      codes(initialState, {
        type: constants.GET_LANGUAGE_CODES_FULFILLED,
        payload: { data: {}, meta: { totalCount: 10 } },
      })
    ).toMatchSnapshot();
  });
  it('should GET_LANGUAGE_CODES_PENDING', () => {
    expect(
      codes(initialState, {
        type: constants.GET_LANGUAGE_CODES_PENDING,
      })
    ).toMatchSnapshot();
  });
  it('should GET_LANGUAGE_CODES_REJECTED', () => {
    expect(
      codes(initialState, {
        type: constants.GET_LANGUAGE_CODES_REJECTED,
        payload: { errors: [{ detail: 'ErrorMessage' }] },
      })
    ).toMatchSnapshot();
  });

  it('should CREATE_LANGUAGE_CODE_FULFILLED', () => {
    expect(
      codes(initialState, {
        type: constants.CREATE_LANGUAGE_CODE_FULFILLED,
        payload: { data: { id: 'id' } },
      })
    ).toMatchSnapshot();
  });
  it('should CREATE_LANGUAGE_CODE_PENDING', () => {
    expect(
      codes(initialState, {
        type: constants.CREATE_LANGUAGE_CODE_PENDING,
      })
    ).toMatchSnapshot();
  });
  it('should CREATE_LANGUAGE_CODE_REJECTED', () => {
    expect(
      codes(initialState, {
        type: constants.CREATE_LANGUAGE_CODE_REJECTED,
        payload: { errors: [{ detail: 'ErrorMessage' }] },
      })
    ).toMatchSnapshot();
  });
});
