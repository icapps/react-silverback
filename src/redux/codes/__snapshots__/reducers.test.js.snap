// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`codes reducer should CREATE_LANGUAGE_CODE_FULFILLED 1`] = `
Object {
  "errorMessage": "",
  "isCreatePending": false,
  "isError": false,
  "isPending": false,
  "languageCode": "id",
  "languageCodeCount": 0,
  "languageCodeList": Array [],
}
`;

exports[`codes reducer should CREATE_LANGUAGE_CODE_PENDING 1`] = `
Object {
  "errorMessage": "",
  "isCreateError": false,
  "isCreatePending": true,
  "isError": false,
  "isPending": false,
  "languageCodeCount": 0,
  "languageCodeList": Array [],
}
`;

exports[`codes reducer should CREATE_LANGUAGE_CODE_REJECTED 1`] = `
Object {
  "errorMessage": "Something went wrong!",
  "isCreateError": true,
  "isCreatePending": false,
  "isError": false,
  "isPending": false,
  "languageCodeCount": 0,
  "languageCodeList": Array [],
}
`;

exports[`codes reducer should GET_LANGUAGE_CODES_FULFILLED 1`] = `
Object {
  "errorMessage": "",
  "isError": false,
  "isPending": false,
  "languageCodeCount": 10,
  "languageCodeList": Object {},
}
`;

exports[`codes reducer should GET_LANGUAGE_CODES_PENDING 1`] = `
Object {
  "errorMessage": "",
  "isError": false,
  "isPending": true,
  "languageCodeCount": 0,
  "languageCodeList": Array [],
}
`;

exports[`codes reducer should GET_LANGUAGE_CODES_REJECTED 1`] = `
Object {
  "errorMessage": "Something went wrong!",
  "isError": true,
  "isPending": false,
  "languageCodeCount": 0,
  "languageCodeList": Array [],
}
`;

exports[`codes reducer should return the initial state 1`] = `
Object {
  "errorMessage": "",
  "isCreateError": false,
  "isCreatePending": false,
  "isError": false,
  "isPending": false,
  "languageCode": null,
  "languageCodeCount": 0,
  "languageCodeList": Array [],
}
`;
